crates/iota-core/src/unit_tests/starfish_manager_tests.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{sync::Arc, time::Duration};
+
+use fastcrypto::traits::KeyPair;
+use iota_metrics::RegistryService;
+use iota_swarm_config::network_config_builder::ConfigBuilder;
+use iota_types::messages_checkpoint::{
+    CertifiedCheckpointSummary, CheckpointContents, CheckpointSummary,
+};
+use prometheus::Registry;
+use tokio::{sync::mpsc, time::sleep};
+
+use crate::{
+    authority::{AuthorityState, test_authority_builder::TestAuthorityBuilder},
+    checkpoints::{CheckpointMetrics, CheckpointService, CheckpointServiceNoop},
+    consensus_handler::ConsensusHandlerInitializer,
+    consensus_manager::{
+        ConsensusManagerMetrics, ConsensusManagerTrait, starfish_manager::StarfishManager,
+    },
+    consensus_validator::{IotaTxValidator, IotaTxValidatorMetrics},
+    starfish_adapter::LazyStarfishClient,
+    state_accumulator::StateAccumulator,
+};
+
+pub fn checkpoint_service_for_testing(state: Arc<AuthorityState>) -> Arc<CheckpointService> {
+    let (output, _result) = mpsc::channel::<(CheckpointContents, CheckpointSummary)>(10);
+    let epoch_store = state.epoch_store_for_testing();
+    let accumulator = Arc::new(StateAccumulator::new_for_tests(
+        state.get_accumulator_store().clone(),
+    ));
+    let (certified_output, _certified_result) = mpsc::channel::<CertifiedCheckpointSummary>(10);
+
+    let (checkpoint_service, _) = CheckpointService::spawn(
+        state.clone(),
+        state.get_checkpoint_store().clone(),
+        epoch_store.clone(),
+        state.get_transaction_cache_reader().clone(),
+        Arc::downgrade(&accumulator),
+        Box::new(output),
+        Box::new(certified_output),
+        CheckpointMetrics::new_for_tests(),
+        3,
+        100_000,
+    );
+    checkpoint_service
+}
+
+#[tokio::test(flavor = "current_thread", start_paused = true)]
+async fn test_starfish_manager() {
+    // GIVEN
```

Comment:
what is given? 😅

Reply:
It's how it was with upstream - GIVEN sets up the environment, WHEN describes behavior/scenario, THEN specifies expected outcome. 

I'd leave it for now as it's simply copied from Mysticeti tests, and cleanups are outside of the scope of this PR. We will definitely be looking into those tests and potentially rewriting them in the future, so I'll keep your suggestion in mind :)